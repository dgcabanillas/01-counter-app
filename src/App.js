import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const App = ({ saludo, subtitulo }) => {
    return (  
        <Fragment>
            <h1>{ saludo }</h1>
            <p>{ subtitulo }</p>
        </Fragment>
    );
}
 
App.propTypes = {
    saludo: PropTypes.string.isRequired
}
App.defaultProps = {
    subtitulo: "Soy un subtitulo"
}

export default App;