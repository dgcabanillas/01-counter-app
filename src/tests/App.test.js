import React from 'react';
//import { render } from '@testing-library/react'
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import App from '../App';

describe('Pruebas en <App />', () => {
    /*
    // -- Ahora usaremos Enzyme
    test('Debe de mostrar el mensaje "Hola, Soy Diego', () => {
        const saludo = "Hola, Soy Diego";
        const { getByText } = render( <App saludo={saludo} /> );
        expect( getByText(saludo) ).toBeInTheDocument();
    });
    */

    test('Debe de mostrar <App /> de manera correcta', () => {
        const saludo = "Hola, Soy Diego";
        const wrapper = shallow( <App saludo={saludo} /> );
        expect( wrapper ).toMatchSnapshot();
    });
    test('Debe mostrar el subtitulo  enviado por props', () => {
        const saludo = 'Hola, Soy Diego';
        const subtitulo = 'Soy un subtitulo';
        const wrapper = shallow( 
            <App 
                saludo={saludo} 
                subtitulo={subtitulo}
            /> 
        );
        const textoParrafo = wrapper.find('p').text();
        expect( textoParrafo ).toBe( subtitulo );
    })

});