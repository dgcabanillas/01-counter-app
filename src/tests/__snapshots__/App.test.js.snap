// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <App /> Debe de mostrar <App /> de manera correcta 1`] = `
<Fragment>
  <h1>
    Hola, Soy Diego
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
